src/main/jbake/content/blog/2017/07/jaeger-with-security-proxy.adoc

```diff
@@ -0,0 +1,253 @@
+  }
+}
+```
+
+And finally, let's create a role and a user, so that we can log into Jaeger's Query service:
+
+* Under the `Configure` left-side menu, open the `Roles` page and click `Add role`
+* As role name, set `user` and click `Save`
+* Under the `Manage` left-side menu, open the `Users` page and click `Add user`
+* Fill out the form as you wish and set `Email verified` to `ON` and click on `Save`
+* Open the `Credentials` tab for this user and set a password (temporary or not).
+* Open the `Role mappings` tab for this user, select the role `user` from the `Available Roles` list and click `Add selected`
+
+=== Preparing OpenShift
+
+For this demo, we assume you have an OpenShift cluster running already. If you don't, then you might want to check out tools
+like link:https://github.com/minishift/minishift[`minishift`]. If you are running a recent version of Fedora, CentOS or
+Red Hat Enterprise Linux you might want to install the package `origin-clients` and run `oc cluster up --version=latest`.
+This should get you a basic OpenShift cluster running locally.
+
+To make it easier for our demonstration, we'll add `cluster-admin` rights to our `developer` user and we'll create 
+the Jaeger namespace:
+
+```bash
+oc login -u system:admin
+oc new-project jaeger
+oc adm policy add-cluster-role-to-user cluster-admin developer
+oc login -u developer
+```
+
+=== Preparing the Jaeger OpenShift template
+
+We'll use the link:https://github.com/jaegertracing/jaeger-openshift/blob/master/production/jaeger-production-template.yml[Jaeger OpenShift Production template]
+as the starting point: either clone the entire repository, or just get a local version of the template.
+
+The first step is to add the sidecar container to the `query-deployment` object. Under the `containers` list, after we
+specify the `jaeger-query`, let's add the sidecar:
+
+```yml
+        - image: jpkroehling/keycloak-proxy
+          name: ${JAEGER_SERVICE_NAME}-query-security-proxy
+          volumeMounts:
+          - mountPath: /opt/jboss/conf
+            name: security-proxy-configuration-volume
+          ports:
+          - containerPort: 8080
+            protocol: TCP
+          readinessProbe:
+            httpGet:
+              path: "/"
+              port: 8080
+```
+
+Note that container specifies a `volumeMount` named `security-proxy-configuration-volume`: we'll use it to store the
+proxy's configuration file. The volume should be added under the `spec` node for the deployment's template and is a sibling
+node to the `dnsPolicy` property:
+
+```yml
+        volumes:
+          - configMap:
+              name: ${JAEGER_SERVICE_NAME}-configuration
+              items:
+                - key: proxy
+                  path: proxy.json
+            name: security-proxy-configuration-volume
+```
+
+Now, we need to specify the `ConfigMap`, with the proxy's configuration entry. To do that, we add a new top-level item to
+the template:
```

Comment:
The template is a collection of items, and this is a new item on this collection. So, anywhere would be fine, as long as it's at the "root" node, ie, part of the collection. I included a suggestion to the reader:

> ... template. As a suggestion, we recommend keeping it close to where it's consumed. For instance, right before the `query-deployment`: